### PR TITLE
Add missing release to ecma_op_container_foreach

### LINKED ISSUE #2910: Assertion JERRY_CONTEXT (jmem_heap_allocated_size) == 0 in jmem_heap_finalize
###### Jerry version:

```
Checked revision: 351acdf5
Build command: ./tools/build.py --clean --debug --error-messages=on --profile=es2015-subset --logging=on
```

###### OS:

```
Linux-4.15.0-51-generic-x86_64-with-Ubuntu-18.04-bionic
```

###### Test case:

```javascript
var map = new Map ; 
for (var i = 0; i < 15 ; i ++ ) { map.set() } 
map.forEach ( Number )
```

###### Backtrace:

```text
ICE: Assertion 'JERRY_CONTEXT (jmem_heap_allocated_size) == 0' failed at jerryscript/jerry-core/jmem/jmem-heap.c(jmem_heap_finalize):143.
Error: ERR_FAILED_INTERNAL_ASSERTION

bt
#0  0xf7fd5059 in __kernel_vsyscall ()
#1  0xf7800832 in raise () from /lib/i386-linux-gnu/libc.so.6
#2  0xf7801cc1 in abort () from /lib/i386-linux-gnu/libc.so.6
#3  0x56579f84 in jerry_port_fatal (code=ERR_FAILED_INTERNAL_ASSERTION) at jerryscript/jerry-port/default/default-fatal.c:71
#4  0x566157a6 in jerry_fatal (code=ERR_FAILED_INTERNAL_ASSERTION) at jerryscript/jerry-core/jrt/jrt-fatals.c:58
#5  0x566157e7 in jerry_assert_fail (assertion=0x566a3760 "JERRY_CONTEXT (jmem_heap_allocated_size) == 0", file=0x566a3700 "jerryscript/jerry-core/jmem/jmem-heap.c", function=0x56678500 <__func__.4545.lto_priv.501> "jmem_heap_finalize", line=143) at jerryscript/jerry-core/jrt/jrt-fatals.c:82
#6  0x56614d6a in jmem_heap_finalize () at jerryscript/jerry-core/jmem/jmem-heap.c:143
#7  0x56614995 in jmem_finalize () at jerryscript/jerry-core/jmem/jmem-allocator.c:170
#8  0x5663deee in jerry_cleanup () at jerryscript/jerry-core/api/jerry.c:231
#9  0x5663bcfb in main (argc=3, argv=0xffffd1b4) at jerryscript/jerry-main/main-unix.c:940
```

<sup>Found by [Fuzzinator](http://fuzzinator.readthedocs.io/) with [grammarinator](https://github.com/renatahodovan/grammarinator). </sup>

### PR DESCRIPTION
**Pull request by szilagyiadam**
The result of the callback function was not freed
Fixes #2910

Co-authored-by: Tibor Dusnoki tdusnoki@inf.u-szeged.hu
JerryScript-DCO-1.0-Signed-off-by: Adam Szilagyi aszilagy@inf.u-szeged.hu

### CODE REVIEW

**Comment by galpeter** [comment]
@szilagyiadam it seems there is something wrong the test fails on the CI.

**Comment by galpeter** [comment]
@szilagyiadam the test is placed in the `tests/jerry/fail/ ` directory indicating that this should fail. However it does not fail (that is why the CI fails). Is this really the place where the test should be?

File: tests/jerry/regression-test-issue-2910.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright JS Foundation and other contributors, http://js.foundation
```
Comment (rerobika) [inline tests/jerry/regression-test-issue-2910.js]: This file must be present only **once** in `tests/jerry/es2015`.

**Comment by galpeter** [comment]
The test file is still in the incorrect place. It should be in the `tests/jerry/es2015` subdir (and not in the `tests/jerry/` dir)

**Comment by szilagyiadam** [comment]
Did the requested changes.

**Pull request merged**